packages/assets-controllers/src/CurrencyRateController.test.ts

```diff
@@ -65,18 +65,10 @@ const getStubbedDate = () => {
   return new Date('2019-04-07T10:20:30Z').getTime();
 };
 
-/**
- * Resolve all pending promises.
- * This method is used for async tests that use fake timers.
- * See https://stackoverflow.com/a/58716087 and https://jestjs.io/docs/timer-mocks.
- */
-function flushPromises(): Promise<unknown> {
-  return new Promise(jest.requireActual('timers').setImmediate);
-}
-
 describe('CurrencyRateController', () => {
   beforeEach(() => {
-    jest.useFakeTimers('legacy');
+    // jest.useFakeTimers({advanceTimers: true});
+    jest.useFakeTimers({doNotFake: ['nextTick', 'setImmediate']});
```

Comment:
so there are the two options for getting nock to play nicely.

1. automatically advance timers (and i believe promises) 20ms every 20ms
2. don't mock `process.nextTick()` and `setImmediate()` which should have the effect of immediately processing promises and any timeouts set via `setImmediate`, and maybe `setTimeout(..., 0)`?